c11 _Static_assert is used in c99 mode

The build fails when linking ``usr/bin/julia``
```
/scratch/julia/usr/lib/libjulia.so: undefined reference to `_Static_assert'
collect2: ld returned 1 exit status
make[1]: *** [/scratch/julia/usr/bin/julia] Error 1
make: *** [julia-ui-release] Error 2
```

I am on 
```
Distributor ID:	CentOS
Release:	6.6

LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
```
with 
```
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    1
Core(s) per socket:    4
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 60
Stepping:              3
CPU MHz:               3301.000
BogoMIPS:              6584.39
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              6144K
NUMA node0 CPU(s):     0-3
```
```
gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-11)
ldd (GNU libc) 2.12
```

--- a/Make.inc
+++ b/Make.inc
@@ -328,7 +328,7 @@ $(error Address Sanitizer only supported with clang. Try setting SANITIZE=0)
 endif
 CC  = icc
 CXX = icpc
-JCFLAGS = -std=gnu99 -pipe $(fPIC) -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -fp-model precise -fp-model except -no-ftz
+JCFLAGS = -std=gnu11 -pipe $(fPIC) -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -fp-model precise -fp-model except -no-ftz
 JCPPFLAGS =
 JCXXFLAGS = -pipe $(fPIC) -fno-rtti
 DEBUGFLAGS = -O0 -g -DJL_DEBUG_BUILD -fstack-protector-all

--- a/src/jl_uv.c
+++ b/src/jl_uv.c
@@ -27,8 +27,23 @@
 #include <io.h>
 #define write _write
 #endif
-#if defined(_COMPILER_MICROSOFT_) && !defined(_Static_assert)
-#define _Static_assert static_assert
+
+#ifndef static_assert
+#  ifdef __cplusplus
+#    if __cplusplus < 201103L
+#      define static_assert(...)
+#    endif
+#  else
+#    define static_assert(...)
+// Remove the following gcc special handling when we officially requires
+// gcc 4.7 (for c++11) and -std=gnu11
+#    ifdef __GNUC__
+#      if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)
+#        undef static_assert
+#        define static_assert _Static_assert
+#      endif
+#    endif
+#  endif
 #endif
 
 #ifdef __cplusplus
@@ -364,7 +379,7 @@ DLLEXPORT void jl_uv_writecb(uv_write_t *req, int status)
 static void jl_write(uv_stream_t *stream, const char *str, size_t n)
 {
     assert(stream);
-    _Static_assert(offsetof(uv_stream_t,type) == offsetof(ios_t,bm) &&
+    static_assert(offsetof(uv_stream_t,type) == offsetof(ios_t,bm) &&
         sizeof(((uv_stream_t*)0)->type) == sizeof(((ios_t*)0)->bm),
 	   "UV and ios layout mismatch");
 

